includes/functions/llms.functions.access.php

```diff
@@ -496,67 +561,46 @@ function llms_is_post_restricted_by_time_period( $post_id, $user_id = null ) {
  */
 function llms_is_post_restricted_by_membership( $post_id, $user_id = null ) {
 
-	// don't check these posts types.
-	$skip = apply_filters(
-		'llms_is_post_restricted_by_membership_skip_post_types',
-		array(
-			'course',
-			'lesson',
-			'llms_quiz',
-			'llms_membership',
-			'llms_question',
-			'llms_certificate',
-			'llms_my_certificate',
-		)
-	);
-
-	if ( in_array( get_post_type( $post_id ), $skip, true ) ) {
-		return false;
-	}
+	$restriction = false;
 
-	$memberships = get_post_meta( $post_id, '_llms_restricted_levels', true );
-	$restricted  = get_post_meta( $post_id, '_llms_is_restricted', true );
-
-	if ( 'yes' === $restricted && $memberships && is_array( $memberships ) ) {
-
-		// if no user, return the first membership from the array as the restriction id.
-		if ( ! $user_id ) {
-
-			$restriction_id = array_shift( $memberships );
+	$memberships = llms_get_post_membership_restrictions( $post_id );
+	if ( $memberships ) {
 
+		$student = $user_id ? llms_get_student( $user_id ) : false;
+		if ( ! $student ) {
+			$restriction = array_shift( $memberships );
```

Comment:
is `reset()` faster than `array_shift()`? Or is `$memberships[0]` faster? As far as I can tell all 3 will do the same thing. What's the advantage to one the other over the other?

Reply:
I think, but I have no benchmarks, that `array_shift() `is slower that `reset()` because PHP doc say that `array_shift()` also performs a `reset()` on the resulting array.
https://www.php.net/manual/en/function.array-shift.php
But maybe on small arrays is not appreciable.
I also think that, since I don't need to reset a pointer of an array nor I need to have a smaller array but just get the first element of it `$memberships[0]` is more appropriate. :)

Reply:
FINE. <3

Reply:
This is called a "heated agreeent"